networkit/cpp/graph/Graph.cpp

```diff
@@ -1065,4 +1068,52 @@ Graph Graph::subgraphFromNodes(const std::unordered_set<node> &nodes) const {
 	return S;
 }
 
+void Graph::toLargestConnectedComponent() {
+	if (this->isDirected())
+		throw std::runtime_error(
+			"Method available for undirected graphs only.");
+
+	ConnectedComponents cc(*this);
+	cc.run();
+	auto comps = cc.getComponents();
+	std::sort(comps.begin(), comps.end(),
+		[&](const std::vector<node> &x, const std::vector<node> &y) {
+			return x.size() > y.size();
+	});
+	std::unordered_map<node, node> map;
+	std::unordered_set<node> set(comps[0].begin(), comps[0].end());
+	node idx = 0;
+	for (node u : comps[0])
+		map[u] = idx++;
+	std::vector<std::pair<node, node>> edges;
+	std::vector<edgeweight> weights;
+	this->subgraphFromNodes(set).forEdges([&](node x, node y, edgeweight w) {
```

Comment:
This creates a copy of the whole graph just to enumerate all edges where both endpoints are in the given set! Why does this not simply iterate over all nodes in the connected component and all of its neighbors and checks if both endpoints are in the largest connected component?